### PR TITLE
feat(integrations): add basic asana integration with pat

### PR DESCRIPTION
**Pull request by pashafateev**

### CODE REVIEW

File: web/static/asana/connect/index.html [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      </li>
+    </ul>
+
+    <!-- TODO(ENG-799): remove ID once its unused -->
+    <form id="save-form" method="post" action="/asana/save">
+      <div class="form-group">
+        <label for="patSecret">PAT (Personal Access Token)</label>
+        <input
+          type="text"
+          name="patSecret"
```
Comment (daabr) [inline web/static/asana/connect/index.html]: Why not just "pat"?

File: integrations/asana/client.go [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+		}
+
+		at := vs.Get(authType)
+		if !at.IsValid() || at.Value() == "" {
+			return sdktypes.NewStatus(sdktypes.StatusCodeWarning, "Init required"), nil
+		}
+
+		if at.Value() == integrations.Init {
+			return sdktypes.NewStatus(sdktypes.StatusCodeOK, "Initialized"), nil
+		}
```
Comment (daabr) [inline integrations/asana/client.go]: It's true that we currently support only PATs, but it looks like they support OAuth apps as well, so it's logical to model this like GitHub.

File: integrations/asana/client.go [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+		req, err := http.NewRequest("GET", "https://app.asana.com/api/1.0/users/me", nil)
+		if err != nil {
+			return sdktypes.InvalidStatus, err
+		}
+
+		ps := vs.Get(patSecret).Value()
+		req.Header.Add("Authorization", "Bearer "+ps)
+
+		client := &http.Client{}
```
Comment (daabr) [inline integrations/asana/client.go]: If you just use `DefaultClient` (a.k.a. `Client{}`), you might as well just use `http.Get()` without a client at all.

File: integrations/asana/client.go [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+	"go.autokitteh.dev/autokitteh/sdk/sdkintegrations"
+	"go.autokitteh.dev/autokitteh/sdk/sdkmodule"
+	"go.autokitteh.dev/autokitteh/sdk/sdkservices"
+	"go.autokitteh.dev/autokitteh/sdk/sdktypes"
+)
+
+type integration struct{ vars sdkservices.Vars }
+
+var (
+	patSecret     = sdktypes.NewSymbol("pat_secret")
```
Comment (daabr) [inline integrations/asana/client.go]: just `pat` - cleaner

File: integrations/asana/save.go [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+		l.Warn("Failed to parse incoming HTTP request", zap.Error(err))
+		c.AbortBadRequest("form parsing error")
+		return
+	}
+
+	// Test the PAT's usability
+	req, err := http.NewRequest("GET", "https://app.asana.com/api/1.0/users/me", nil)
+	if err != nil {
+		l.Warn("Failed to create HTTP request", zap.Error(err))
+		c.AbortBadRequest("request creation error")
```
Comment (daabr) [inline integrations/asana/save.go]: Please distinguish between bad request errors, and internal server errors (where the error probably isn't caused by the details of the request)

File: integrations/asana/client.go (hunk 4 above)
Comment (pashafateev) [reply integrations/asana/client.go]: Done

File: integrations/asana/client.go (hunk 2 above)
Comment (pashafateev) [reply integrations/asana/client.go]: Done

File: integrations/asana/save.go (hunk 5 above)
Comment (pashafateev) [reply integrations/asana/save.go]: Done

File: web/static/asana/connect/index.html (hunk 1 above)
Comment (pashafateev) [reply web/static/asana/connect/index.html]: Done

File: integrations/asana/save.go [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+		return
+	}
+
+	req.Header.Add("Authorization", "Bearer "+ps)
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		l.Error("Failed to execute HTTP request", zap.Error(err))
+		c.AbortServerError("execution error")
```
Comment (daabr) [inline integrations/asana/save.go]: This includes auth errors, which are more client errors than server errors

**Review by daabr — APPROVED** [review]

File: integrations/asana/save.go (hunk 6 above)
Comment (pashafateev) [reply integrations/asana/save.go]: Done

**Pull request merged**